test/functional/unified-spec-runner/match.ts

```diff
@@ -129,10 +129,7 @@ export function resultCheck(
 
       if (depth > 1) {
         expect(actual, `Expected actual to exist at ${path.join('')}`).to.exist;
-        expect(
-          Object.keys(actual),
-          `[${Object.keys(actual)}] length !== [${Object.keys(expected)}]`
-        ).to.have.lengthOf(Object.keys(expected).length);
+        expect(Object.keys(actual)).to.include.members(Object.keys(expected));
```

Comment:
https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#allowing-extra-fields-in-root-level-documents

Unfortunately this is part of the enforcement of the unified spec, is there a test in particular thats failing if you reverse this?
Or perhaps my depth check isn't correctly filtering out the root level document?

Reply:
No, your code was correct, thanks for pointing this out. After a second look, [this line](https://github.com/mongodb/node-mongodb-native/blob/4.0/src/operations/update.ts#L277) was causing some spec tests to fail due to a presence of `multi: false`, because `multi` defaults to false and should only be passed if true.